controllers/backup.go

```diff
@@ -41,8 +41,12 @@ func (r *ClusterGroupUpgradeReconciler) reconcileBackup(
 			// Backup is done
 			return nil
 		}
+
 		// Backup is required and not marked as done
-		return r.triggerBackup(ctx, clusterGroupUpgrade)
+		if *clusterGroupUpgrade.Spec.Enable {
```

Comment:
this is also redundant now that reconcileBackup is only called after enable and blocking CR check.